### PR TITLE
3714 pandadoc components

### PR DESCRIPTION
**Pull request by alysonturing**
resolves: #3714 

Could not create sources because PandaDoc`s webhooks are only available to enterprise plan as [you can see here](https://support.pandadoc.com/hc/en-us/articles/360007915853--Webhooks)

### CODE REVIEW

File: components/pandadoc/pandadoc.app.mjs [hunk 1]
```diff
@@ -1,11 +1,145 @@
...
+        ...args,
+      });
+    },
+    listDocumentFolders(args = {}) {
+      return this.makeRequest({
+        path: "/documents/folders",
+        ...args,
+      });
+    },
+    documentDetails(args = {}) {
```
Comment (GTFalcao) [inline components/pandadoc/pandadoc.app.mjs:105]: This method is not being used anywhere

**Review by GTFalcao — COMMENTED** [review]
Hi @alysonturing , thanks for your contribution!

I left a few comments, please check them out.

File: components/pandadoc/actions/create-or-update-contact/create-or-update-contact.mjs [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import app from "../../pandadoc.app.mjs";
+
+export default {
+  key: "pandadoc-create-or-update-contact",
+  name: "Create or Update Contact",
+  description: "This method adds or updates a contact using the email as index. [See the docs here](https://developers.pandadoc.com/reference/update-contact)",
```
[the remark this answers is not in the document]
Comment (alysonturing) [reply components/pandadoc/actions/create-or-update-contact/create-or-update-contact.mjs:6]: this method updates when the user is found with the same email, as described on the description

**Review by GTFalcao — APPROVED** [review]
Looks good to me!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[PandaDoc_3714.pdf](https://github.com/PipedreamHQ/pipedream/files/10927074/PandaDoc_3714.pdf)

**Pull request merged**